### DISCUSSION TITLE
Problems configuring the fan modes

### DISCUSSION THREAD

**Discussion by FaGerstenlauer** [echavet/MitsubishiCN105ESPHome#117] [Q&A]
With the great suport of Eric Chavet I got my ESP32 to work. (see [https://github.com/echavet/MitsubishiCN105ESPHome/discussions/110](url))

I have seen that my system do not support "only fan".

![grafik](https://github.com/echavet/MitsubishiCN105ESPHome/assets/148796483/8c59d8b3-1331-4769-b031-c81ac45f02ec)

So in the code I wrote:
```
supports:
      mode: [AUTO, COOL, DRY, HEAT]
```
And it works perfectly.

Now I have the problem that I don't understand how to configure the fan mode. The system suports the folowing:

![grafik](https://github.com/echavet/MitsubishiCN105ESPHome/assets/148796483/2963f88e-c3da-4893-9858-acdd0b0e539a)

What do I have to write in the code? There is no "Super High". If I konfigure it like this:
`      fan_mode: [AUTO, QUIET, LOW, MEDIUM, HIGH]`
If I now switch throuch the modes with the original remote, HA interprets it like that:

Auto  -->  Auto
Silent  -->  Low (wrong, it should be "quiet")
Low  -->  Low
Med.  -->  Middle
High  -->  (error masage: The fan mode is invalid. Valide fan modes are: auto, low, medium, high, quiet)
Super High  -->  High

I'm a bit confused. What do I have to set?

**Comment by disruptivepatternmaterial** [comment]
DEFAULT_FAN_MODES = ["AUTO", "MIDDLE", "QUIET", "LOW", "MEDIUM", "HIGH"]

Those are the 6 modes, you have 6 modes listed: map them as you like.

**Comment by FaGerstenlauer** [comment]
And how do I map them?
